projects/ReOxide/programs/reoxide/module.nix

```diff
@@ -0,0 +1,22 @@
+{
+  config,
+  lib,
+  pkgs,
+  ...
+}:
+let
+  cfg = config.programs.reoxide;
+in
+{
+  options.programs.reoxide = {
+    enable = lib.mkEnableOption "enable reoxide";
+    package = lib.mkPackageOption pkgs "reoxide" { };
+  };
+
+  config = lib.mkIf cfg.enable {
+    environment.systemPackages = with pkgs; [
+      cfg.package
+      reoxide-plugin-simple
```

Comment:
Installing the plugin to `systemPackages` is probably not that useful, since users won't be able to consume it.

Do you think it would be possible to wrap reoxide with the plugins it needs?

Reply:
> I think, however, that we can have a module option in the service, to specify plugins that will be loaded.

That would probably be the best course of action.

> This would still need packaging of the plugins, though.

Yeah, that's fine. We can make a package set that contains all the plugins (for example, see the [bigbluebutton scope](https://github.com/ngi-nix/ngipkgs/blob/45f16613865ba9cc9e4578f518f1a33051f54069/pkgs/by-name/bigbluebutton/sbt-derivation.nix)).